src/components/OpportunityDetails/OpportunityDetails.tsx

```diff
@@ -36,19 +39,23 @@ export interface OpportunityDetailsProps {
  */
 const OpportunityDetails = ({ data, isLoggedIn, tyyppi, sections, showAiInfoInTitle }: OpportunityDetailsProps) => {
   const { isDev } = useEnvironment();
-  const [loginModalOpen, setLoginModalOpen] = React.useState(false);
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const title = getLocalizedText(data?.otsikko);
   const toolStore = useToolStore();
   const [isFavorite, setIsFavorite] = React.useState(false);
+  const { showDialog } = useModal();
+  const state = useLocation().state;
+  const loginLink = useLoginLink({
+    callbackURL: state?.callbackURL
+      ? `/${i18n.language}/${state?.callbackURL}`
+      : `/${i18n.language}/${t('slugs.profile.index')}/${t('slugs.profile.front')}`,
+  });
 
   React.useEffect(() => {
-    setIsFavorite(toolStore.suosikit?.some((suosikki) => suosikki.kohdeId === data?.id));
-  }, [toolStore.suosikit, data?.id]);
-
-  const handleLoginRequired = () => {
-    setLoginModalOpen(true);
-  };
+    if (isLoggedIn && data?.id) {
+      setIsFavorite(toolStore.suosikit?.some((suosikki) => suosikki.kohdeId === data?.id));
+    }
+  }, [toolStore.suosikit, data?.id, isLoggedIn]);
 
```

Comment:
Is this effect necessary, looks like one could compute isFavorite directly from the props and toolStore without keeping additional state?